internal/nix/store.go

```diff
@@ -0,0 +1,42 @@
+package nix
+
+import (
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+func StorePath(hash, name, version string) string {
+	storeDirParts := []string{hash, name}
+	if version != "" {
+		storeDirParts = append(storeDirParts, version)
+	}
+	storeDir := strings.Join(storeDirParts, "-")
+	return filepath.Join("/nix/store", storeDir)
+}
+
+// contentAddressedRegex matches the output of `nix store make-content-addressed`.
+// It is used to select the content-addressed store path (the second one in the example below).
+//
+// Example:
+// > nix store make-content-addressed  /nix/store/r2jd6ygnmirm2g803mksqqjm4y39yi6i-git-2.33.1
+// rewrote '/nix/store/r2jd6ygnmirm2g803mksqqjm4y39yi6i-git-2.33.1' to '/nix/store/ldbhlwhh39wha58rm61bkiiwm6j7211j-git-2.33.1'
+var contentAddressedRegex = regexp.MustCompile(`rewrote\s'[\/a-z0-9-\.]+'\sto\s'([a-z0-9-\/\.]+)'`)
```

Comment:
Does `nix store make-content-addressed --json` work?

Reply:
oh yeah, good call. Will try that...